### PR TITLE
Add PWA manifest generation

### PR DESCRIPTION
**Pull request by Stanzilla**
As a potential successor to https://github.com/squidfunk/mkdocs-material/pull/1405, starting with just adding a PWA manifest. Would love to hear options about how to handle color and icons.

### CODE REVIEW

File: material/manifest.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "mkdocs-material",
```
Comment (squidfunk) [inline material/manifest.json:2]: As I understand, everbody who will be using the theme will have these values set? Wasn't the fact that this is not configurable the reason why we rejected a native PWA integration in the first place?

Comment (Stanzilla) [reply material/manifest.json:2]: Yeah! So, I mainly put that as a placeholder for now, need to figure out a way to get the config values from mkdocs.yml into here. If that is not at all possible, then I need to rethink.

Comment (squidfunk) [reply material/manifest.json:2]: A plugin may be a better approach than to encode this into the theme, as you would have to jump through a lot of hoops to extra information from `mkdocs.yml` in anything other than `*.html` and `*.xml` files.

**Comment by squidfunk** [comment]
Is this still relevant? Would you like to re-open this PR when you managed to address the issues discussed?

**Comment by Stanzilla** [comment]
Yeah, third time might be the charm then :p

**Pull request closed without merging**